### PR TITLE
feat(progress-tracker): add component and demo file

### PR DESCRIPTION
**Pull request by mcastagnetti**
Adds the progress tracker component

https://lumapps.gitbook.io/design-system/progress-tracker

### CODE REVIEW

File: src/components/progress-tracker/react/ProgressTracker.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    className = '',
+    theme = DEFAULT_PROPS.theme,
+    ...props
+}: ProgressTrackerProps): React.ReactElement => {
+    const backgroundPosition: number = 100 / (children.length * 2);
+    const trackPosition: number = ((100 / (children.length - 1)) * activeStep) / 100;
+
+    return (
+        <div className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, theme }))} {...props}>
+            <div className="lumx-progress-tracker__steps" ng-transclude>
```
Comment (GuillaumeNachury) [inline src/components/progress-tracker/react/ProgressTracker.tsx]: You could have used `${CLASSNAME}__steps` for the classname.
Also, there is `ng-transclude` that sneaked in :)

File: src/components/progress-tracker/react/ProgressTracker.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                {children}
+            </div>
+
+            <div
+                className="lumx-progress-tracker__background-bar"
+                style={{ left: `${backgroundPosition}%`, right: `${backgroundPosition}%` }}
+            />
+
+            <div
+                className="lumx-progress-tracker__foreground-bar"
```
Comment (GuillaumeNachury) [inline src/components/progress-tracker/react/ProgressTracker.tsx]: You could have used `${CLASSNAME}__foreground-bar` for the classname.

File: src/components/progress-tracker/react/ProgressTracker.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    const trackPosition: number = ((100 / (children.length - 1)) * activeStep) / 100;
+
+    return (
+        <div className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, theme }))} {...props}>
+            <div className="lumx-progress-tracker__steps" ng-transclude>
+                {children}
+            </div>
+
+            <div
+                className="lumx-progress-tracker__background-bar"
```
Comment (GuillaumeNachury) [inline src/components/progress-tracker/react/ProgressTracker.tsx]: You could have used `${CLASSNAME}__background-bar` for the classname.

File: src/components/progress-tracker/react/ProgressTrackerStep.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                    theme,
+                }),
+                { [`${CLASSNAME}--has-error`]: hasError },
+            )}
+            tabIndex={isClickable ? 0 : -1}
+            onClick={onClick}
+            onKeyDown={onEnterPressed(onClick)}
+            {...restProps}
+        >
+            <Icon className="lumx-progress-tracker-step__state" icon={getIcon()} size={Sizes.s} />
```
Comment (GuillaumeNachury) [inline src/components/progress-tracker/react/ProgressTrackerStep.tsx]: You could have used `${CLASSNAME}__state` for the classname.

File: src/components/progress-tracker/react/ProgressTrackerStep.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+            <span className="lumx-progress-tracker-step__label">{label}</span>
+
+            {helper && <span className="lumx-progress-tracker-step__helper">{helper}</span>}
```
Comment (GuillaumeNachury) [inline src/components/progress-tracker/react/ProgressTrackerStep.tsx]: You could have used `${CLASSNAME}__helper` for the classname.

File: src/components/progress-tracker/react/ProgressTracker.tsx (hunk 1 above)
Comment (mcastagnetti) [reply src/components/progress-tracker/react/ProgressTracker.tsx]: Lol, a nice copy pasta :angel:

**Comment by mcastagnetti** [comment]
Changes done

**Comment by matmkian** [comment]
Can't test an active step for example. More globally, is it possible for you to set the same demo page as the one in angularjs version -> so that I can test every case?

**Comment by matmkian** [comment]
Test ok

**Pull request merged**